DTS overlay files must include full path name

CMake handles overlay files inconsistently. My sample directory contains an overlay file:
-rw-rw-r-- 1 pflykt pflykt 204 Nov  8 15:11 frdm_uart2_dts.overlay

But when I run cmake, cc1 does not find the DTS overlay file:
```
(samples_fona808)$ cmake -B build -GNinja -DBOARD=frdm_k64f -DDTC_OVERLAY_FILE=frdm_uart2_dts.overlay
Zephyr version: 2.0.99
-- Selected BOARD frdm_k64f
-- Found west: /home/pflykt/.local/bin/west (found suitable version "0.6.0", minimum required is "0.6.0")
-- Loading /home/pflykt/projects/zephyr/boards/arm/frdm_k64f/frdm_k64f.dts as base
-- Overlaying frdm_uart2_dts.overlay
cc1: fatal error: frdm_uart2_dts.overlay: No such file or directory
compilation terminated.
CMake Error at /home/pflykt/projects/zephyr/cmake/dts.cmake:128 (message):
  command failed with return code: 1
Call Stack (most recent call first):
  /home/pflykt/projects/zephyr/cmake/app/boilerplate.cmake:460 (include)
  CMakeLists.txt:7 (include)
```
Adding the full path to the command succeeds:
```
samples_fona808)$ cmake -B build -GNinja -DBOARD=frdm_k64f -DDTC_OVERLAY_FILE=$(pwd)/frdm_uart2_dts.overlay
Zephyr version: 2.0.99
-- Selected BOARD frdm_k64f
-- Found west: /home/pflykt/.local/bin/west (found suitable version "0.6.0", minimum required is "0.6.0")
-- Loading /home/pflykt/projects/zephyr/boards/arm/frdm_k64f/frdm_k64f.dts as base
-- Overlaying /home/pflykt/projects/zephyr/samples/fona808/frdm_uart2_dts.overlay
Devicetree configuration written to /home/pflykt/projects/zephyr/samples/fona808/build/zephyr/include/generated/generated_dts_board.conf
Parsing Kconfig tree in /home/pflykt/projects/zephyrproject/zephyr/Kconfig
Loaded configuration '/home/pflykt/projects/zephyr/boards/arm/frdm_k64f/frdm_k64f_defconfig'
...
```
The same thing happens when `DTC_OVERLAY_FILE` is defined in `CMakeLists.txt`.

Documentation in https://docs.zephyrproject.org/latest/application/index.html#devicetree-overlays is very quiet about including the full path name, and the above handling of DTS overlay files shows inconsistent behavior against project overlay files. All overlay files must be handled in a consistent manner, please fix.

--- a/cmake/dts.cmake
+++ b/cmake/dts.cmake
@@ -126,10 +126,10 @@ if(SUPPORTS_DTS)
     -P
     -E   # Stop after preprocessing
     -MD  # Generate a dependency file as a side-effect
-    -MF ${BOARD}.dts.pre.d
-    -o ${BOARD}.dts.pre.tmp
+    -MF ${PROJECT_BINARY_DIR}/${BOARD}.dts.pre.d
+    -o  ${PROJECT_BINARY_DIR}/${BOARD}.dts.pre.tmp
     ${ZEPHYR_BASE}/misc/empty_file.c
-    WORKING_DIRECTORY ${PROJECT_BINARY_DIR}
+    WORKING_DIRECTORY ${APPLICATION_SOURCE_DIR}
     RESULT_VARIABLE ret
     )
   if(NOT "${ret}" STREQUAL "0")

